test-framework/test-suites/integration/tests/remove/test_remove_pallet.py

```diff
@@ -231,3 +231,44 @@ def test_multiple_versions(self, host, create_pallet_isos, revert_etc, revert_ex
 
 		# But the '1.0' version of the pallet should be gone
 		assert not host.file('/export/stack/pallets/minimal/1.0').exists
+
+	def test_remove_partially_removed_pallet(self, host, create_pallet_isos, revert_export_stack_pallets):
+		# Add our minimal pallet
+		result = host.run(f'stack add pallet {create_pallet_isos}/minimal-1.0-sles12.x86_64.disk1.iso')
+		assert result.rc == 0
+
+		# Make sure it made it to the DB
+		result = host.run('stack list pallet minimal output-format=json')
+		assert result.rc == 0
+		assert json.loads(result.stdout) == [
+			{
+				'name': 'minimal',
+				'version': '1.0',
+				'release': 'sles12',
+				'arch': 'x86_64',
+				'os': 'sles',
+				'boxes': ''
+			}
+		]
+
+		# Sanity check that the pallet files got added to the filesystem
+		assert host.file('/export/stack/pallets/minimal/1.0/sles12/sles/x86_64/roll-minimal.xml').exists
+
+		# now kill part of the tree
+		shutil.rmtree('/export/stack/pallets/minimal/1.0/sles12/sles/x86_64/')
+		assert not host.file('/export/stack/pallets/minimal/1.0/sles12/sles/x86_64/roll-minimal.xml').exists
+
+		# Make sure it is still in the DB
+		result = host.run('stack list pallet minimal output-format=json')
+		assert result.rc == 0
+		assert len(json.loads(result.stdout)) == 1
```

Comment:
Why is this not performing an equality check against a list of dictionaries like the first testing of list pallet above?